docker-host/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/robfig/cron/v3"
+	machineryvars "github.com/uselagoon/machinery/utils/variables"
+)
+
+const EnvOverrideHost = "DOCKER_HOST"
+
+var dockerHost = machineryvars.GetEnv("DOCKER_HOST", "docker-host")
+var repo = machineryvars.GetEnv("REPOSITORY_TO_UPDATE", "amazeeio")
+var REGISTRY = machineryvars.GetEnv("REGISTRY", "docker-registry.default.svc:5000")
+var BIP = machineryvars.GetEnv("BIP", "172.16.0.1/16")
+var REGISTRY_MIRROR = machineryvars.GetEnv("REGISTRY_MIRROR", "https://imagecache.amazeeio.cloud")
+var pruneImagesSchedule = machineryvars.GetEnv("PRUNE_SCHEDULE", "22 1 * * *")
+var removeExitedSchedule = machineryvars.GetEnv("REMOVE_EXITED_SCHEDULE", "22 */4 * * *")
+var updateImagesSchedule = machineryvars.GetEnv("UPDATE_IMAGES_SCHEDULE", "*/15 * * * *")
+var pruneImagesUntil = machineryvars.GetEnv("PRUNE_IMAGES_UNTIL", "168h")
+var danglingFilter = machineryvars.GetEnv("DANGLING_FILTER", "true")
+
+func main() {
+	cli, err := client.NewClientWithOpts(
+		client.WithHostFromEnv(),
+		client.WithAPIVersionNegotiation(),
+	)
+	if err != nil {
+		fmt.Println("Error", err)
+	}
+	defer cli.Close()
+
+	var command = fmt.Sprintf("/usr/local/bin/dind /usr/local/bin/dockerd --host=tcp://0.0.0.0:2375 --host=unix:///var/run/docker.sock --insecure-registry=%s --insecure-registry=harbor-harbor-core.harbor.svc.cluster.local:80 --bip=%s --storage-driver=overlay2 --storage-opt=overlay2.override_kernel_check=1 --registry-mirror=%s", REGISTRY, BIP, REGISTRY_MIRROR)
```

Comment:
Should break this command out to be incrementally created based on what options are defined.

Ideally the `--registry-mirror=%s` option should only be added if `REGISTRY_MIRROR` is populated. We want to deprecate coding in the `imagecache.amazeeio.cloud` value in places. So setting the `--registry-mirror` flag will need to be optional in the future.

Additionally the `--insecure-registry=%s` option should also be the same, only set if `REGISTRY` is populated. As it is possible to start the docker-host without defining this registry.

I also think we can remove this flag entirely as this won't be used anywhere `--insecure-registry=harbor-harbor-core.harbor.svc.cluster.local:80`